src/main/scala/com/lightbend/sbt/MultiReleaseJarPlugin.scala

```diff
@@ -35,19 +35,69 @@ object MultiReleaseJarPlugin extends AutoPlugin {
   val MULTI_RELEASE_KEY = "Multi-Release"
   // ---------------------------------------
 
-  override def projectSettings = Seq(
-    
-    compile := 
+  def jdkVersion: String = System.getProperty("java.version")
+  
+  override def projectSettings: Seq[Def.Setting[_]] =
+    jdkVersion match {
+      case "9" =>
+        println(
+          scala.Console.BOLD + 
+            "[multi-release-jar plugin] Running using JDK9! " +
+            "Will include classes and tests that require JDK9." +
+            scala.Console.RESET)
+        jdk9ProjectSettings
+      case "1.8" | "8" => 
+        println(
+          scala.Console.BOLD + 
+            "[multi-release-jar plugin] Running using JDK 8, " +
+            "note that JDK9 classes and tests will not be included in compile/test runs." + 
+            scala.Console.RESET)
+        
+        Seq.empty
+      case _ => 
+        throw new IllegalStateException("Only JDK 8 or 9 is supported by this build, because of the mult-release-jar plugin.")
+    }
+  
+  def jdk9ProjectSettings: Seq[Def.Setting[_]] = Seq(
+    compile in Compile := 
       (compile in MultiReleaseJar).dependsOn(
         compile in Compile
       ).value,
     
+//    test := 
+//      (test in MultiReleaseJarTest).dependsOn(
+//        test in Test
+//      ).value,
+    
     Keys.`package` :=
       (Keys.`package` in Compile).dependsOn(
         compile in Compile, 
         compile in MultiReleaseJar
       ).value
     
+  ) ++ Seq(
+    // ----    testing configuration     ----
+    sourceDirectories in Test ++= {
```

Comment:
I'm somewhat confused why this "works" but "has no effect".

If I show test:SourceDirectories I get:

```
[info] [info] Loading settings from build.sbt ...
[info] [info] Set current project to jdk9-testing-test (in build file:/private/var/folders/6w/p7yjvbwd14v4jh29ytjsq11m0000gn/T/sbt_b9ac51af/jdk9-testing-test/)
[info] [info] * /private/var/folders/6w/p7yjvbwd14v4jh29ytjsq11m0000gn/T/sbt_b9ac51af/jdk9-testing-test/src/test/scala-2.12
[info] [info] * /private/var/folders/6w/p7yjvbwd14v4jh29ytjsq11m0000gn/T/sbt_b9ac51af/jdk9-testing-test/src/test/scala
[info] [info] * /private/var/folders/6w/p7yjvbwd14v4jh29ytjsq11m0000gn/T/sbt_b9ac51af/jdk9-testing-test/src/test/java
[info] [info] * /private/var/folders/6w/p7yjvbwd14v4jh29ytjsq11m0000gn/T/sbt_b9ac51af/jdk9-testing-test/target/scala-2.12/src_managed/test
[info] [info] * /private/var/folders/6w/p7yjvbwd14v4jh29ytjsq11m0000gn/T/sbt_b9ac51af/jdk9-testing-test/src/test/java-jdk9
[info] [info] * /private/var/folders/6w/p7yjvbwd14v4jh29ytjsq11m0000gn/T/sbt_b9ac51af/jdk9-testing-test/src/test/scala-jdk9
```

Which should do the trick I thought, but it does not attempt to compile the source in that directory at all hmmm

Reply:
Which effect did you expect and what did happen instead?

Reply:
Not only should it be included in that list, but those files (in those directories) should also be compiled is what I expected. If classpath will end up wrong I can fix that, but I don't get why it's listing the dirs correctly, but not compiling things from them - specifically the `jdk9-testing-test/src/test/scala-jdk9/com/example/Jdk9VersionOfSomeTests.scala`

Reply:
I guess it's because this setting is not used for compiling ;)

I think it should be `managedSourceDirectories` instead.

Reply:
Eh, I mean `unmanagedSourceDirectories` if the user puts files into it.

Reply:
I guess one reason why that's so hard to find out is that `inspect tree` "misses" some edges when the keys delegate to something else. This can be configured by passing `actual = true` to `Def.compiled` which `inspect` does but `inspect tree` does not so it's hard to trace back the actual dependency path from `compileIncremental` to `unmanagedSourceDependencies`.

I built a better `inspect2 tree` (subtle naming, huh?) that will follow the delegations as well. Can be copied to any project for now from here: https://gist.github.com/jrudolph/9e0e0e8a0df35c23713dfc314670db87

Reply:
While at it I went a step further and implemented an `inspect2 how` command that shows paths from a goal setting to a transitive dependency. E.g. for `test:compile` and `test:unmanagedSourceDirectories`:

```
> inspect2 how test:compile test:unmanagedSourceDirectories
[info] backend/test:compile -> backend/test:manipulateBytecode -> backend/test:compileIncremental -> backend/test:compile::compileInputs -> backend/test:sources -> backend/test:unmanagedSources -> backend/test:unmanagedSourceDirectories
[info] backend/test:compile -> backend/test:manipulateBytecode -> backend/test:compileIncremental -> backend/test:compile::compileInputs -> backend/test:scalariformFormat -> backend/test:sourceDirectories -> backend/test:unmanagedSourceDirectories
```

If the output is empty, the two settings don't relate.

Reply:
StackOverflow answers lied to me, damn them! :) (I guess it was about some ancient version...)
Indeed the unmanaged seems to do the trick, thanks